src/sentry/web/frontend/accounts.py

```diff
@@ -74,20 +74,27 @@ def expired(request, user):
 def recover(request):
     from sentry.app import ratelimiter
 
+    extra = {'ip_address': request.META['REMOTE_ADDR']}
     if request.method == 'POST' and ratelimiter.is_limited(
-        'accounts:recover:{}'.format(request.META['REMOTE_ADDR']),
+        'accounts:recover:{}'.format(extra['ip_address']),
         limit=5, window=60,  # 5 per minute should be enough for anyone
     ):
         return HttpResponse(
             'You have made too many password recovery attempts. Please try again later.',
             content_type='text/plain',
             status=429,
         )
+        logger.warning('recover.rate-limited', extra=extra)
 
     form = RecoverPasswordForm(request.POST or None)
```

Comment:
Why's it recursive? You mean because it's used for both GET and POST? Just do like:

```python
if request.method == 'POST':
  logger.warning('recover.attempt', ...)
```